tests/Microsoft.Data.Analysis.Tests/ArrayComparer.cs

```diff
@@ -54,6 +55,24 @@ public ArrayComparer(IArrowArray expectedArray)
         public void Visit(BinaryArray array) => throw new NotImplementedException();
         public void Visit(IArrowArray array) => throw new NotImplementedException();
 
+        public void Visit(StructArray array)
+        {
+            Assert.IsAssignableFrom<StructArray>(_expectedArray);
+            StructArray expectedArray = (StructArray)_expectedArray;
+
+            Assert.Equal(expectedArray.Length, array.Length);
+            Assert.Equal(expectedArray.NullCount, array.NullCount);
+            Assert.Equal(expectedArray.Offset, array.Offset);
+            Assert.Equal(expectedArray.Data.Children.Length, array.Data.Children.Length);
+
+            var arrayFieldsEnumerator = array.Fields.GetEnumerator();
```

Comment:
We should be comparing the 2 `Fields.Count` are equal. If `array.Fields` is less, the while loop below will just skip the `expectedArray.Fields`.

Check out how we do it in the Arrow tests:

[https://github.com/apache/arrow/blob/04660f8afbf5bee052f8881b36b5c59a8c8ca7b5/csharp/test/Apache.Arrow.Tests/ArrowReaderVerifier.cs#L120-L125](https://github.com/apache/arrow/blob/04660f8afbf5bee052f8881b36b5c59a8c8ca7b5/csharp/test/Apache.Arrow.Tests/ArrowReaderVerifier.cs#L120-L125)

Reply:
Done